set.mm

```diff
@@ -775935,4 +775936,303 @@ to Davis and Putnam (1960).  (Contributed by David A. Wheeler,
+$(
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+  Mathbox for Zhi Wang
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+$)
+
+
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Propositional calculus
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  ${
+    pm4.71da.1 $e |- ( ph -> ( ps <-> ch ) ) $.
+    $( Deduction converting a biconditional to a biconditional with
+       conjunction.  Variant of ~ pm4.71d .  (Contributed by Zhi Wang,
+       30-Aug-2024.) $)
+    pm4.71da $p |- ( ph -> ( ps <-> ( ps /\ ch ) ) ) $=
+      ( biimpd pm4.71d ) ABCABCDEF $.
+
+    ${
+      logic1.2 $e |- ( ph -> ( ps -> ( th <-> ta ) ) ) $.
+      $( Distribution of implication over biconditional with replacement
+         (deduction form).  (Contributed by Zhi Wang, 30-Aug-2024.) $)
+      logic1 $p |- ( ph -> ( ( ps -> th ) <-> ( ch -> ta ) ) ) $=
+        ( wi pm5.74d imbi1d bitrd ) ABDHBEHCEHABDEGIABCEFJK $.
+    $}
+
+    ${
+      logic1a.2 $e |- ( ( ph /\ ps ) -> ( th <-> ta ) ) $.
+      $( Variant of ~ logic1 .  (Contributed by Zhi Wang, 30-Aug-2024.) $)
+      logic1a $p |- ( ph -> ( ( ps -> th ) <-> ( ch -> ta ) ) ) $=
+        ( wb ex logic1 ) ABCDEFABDEHGIJ $.
+    $}
+
+    ${
+      logic2.2 $e |- ( ph -> ( ( ps /\ ch ) -> ( th <-> ta ) ) ) $.
+      $( Variant of ~ logic1 .  (Contributed by Zhi Wang, 30-Aug-2024.) $)
+      logic2 $p |- ( ph -> ( ( ps -> th ) <-> ( ch -> ta ) ) ) $=
+        ( wa wb pm4.71da sylbid logic1 ) ABCDEFABBCHDEIABCFJGKL $.
+    $}
+  $}
+
+  ${
+    pm5.32dav.1 $e |- ( ( ph /\ ps ) -> ( ch <-> th ) ) $.
+    $( Distribution of implication over biconditional (deduction form).
+       Variant of ~ pm5.32da .  (Contributed by Zhi Wang, 30-Aug-2024.) $)
+    pm5.32dav $p |- ( ph -> ( ( ch /\ ps ) <-> ( th /\ ps ) ) ) $=
+      ( wa pm5.32da ancom 3bitr3g ) ABCFBDFCBFDBFABCDEGBCHBDHI $.
+  $}
+
+
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  ZF Set Theory - start with the Axiom of Extensionality
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Restricted quantification
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  ${
+    $d ph x $.
+    ralbidb.1 $e |- ( ph -> ( x e. A <-> ( x e. B /\ ps ) ) ) $.
+    ralbidb.2 $e |- ( ph -> ( ( x e. A /\ ( x e. B /\ ps ) )
+                           -> ( ch <-> th ) ) ) $.
+    $( Formula-building rule for restricted universal quantifier and additional
+       condition (deduction form).  (Contributed by Zhi Wang, 30-Aug-2024.) $)
+    ralbidb $p  |- ( ph -> ( A. x e. A ch <-> A. x e. B ( ps -> th ) ) ) $=
+      ( wi cv wcel wa logic2 impexp bitrdi ralbidv2 ) ACBDJZEFGAEKZFLZCJSGLZBMZ
+      DJUARJATUBCDHINUABDOPQ $.
+  $}
```

Comment:
Since `( x e. A <-> ( x e. B /\ ps ) )`, maybe `ralbidb.2` can be simplified to `( ( ph /\ x e. A ) -> ( ch <-> th ) )` ?

Reply:
Exactly. I thought so but without `ralbidb.1`, ``( ( ph /\ x e. A ) -> ( ch <-> th ) )`` would be a stronger hypothesis than ``ralbidb.2``?

Reply:
There's not really any guideline about theorem generality, but here's a description of things that happen kinda by default as a consequence of natural heuristics and move-to-main guidelines:

If the extra generality is used then keep it, if it's interesting to you keep it (usually mathbox until moved), and if you're proving something balance generality with ease of proof and what you're using it for. Sometimes if you proved a general thing naturally and don't need it, it's better/easier to just prove the more specific from the general, and other times the general statement is revised.

Since it is mathbox, it is your call, but by default it probably isn't needed or interesting. I don't know how this theorem is used, but maybe the simpler hypothesis is easier to use? This is non-blocking.

Reply:
Thanks for the comments. I fixed this in the latest commits.